### ISSUE TITLE
DELETE /v2/apps/{appId} seems inconsistent

### ISSUE THREAD

**Issue by jolexa** [mesosphere/marathon#3225]
```
DELETE /v2/apps/{appId}

Destroy an application. All data about that application will be deleted.
```

Hello,
I have an app that respects SIGTERM and gracefully terminates. In my observations, when I issue a `kill` from the Marathon UI, it seems a SIGTERM is sent. :+1: However, when I use DELETE against the Marathon API, it seems a SIGKILL is sent.

Can this be right?

**Comment by philipnrmn** [comment]
Hi @jolexa,

There's a small distinction here, the `kill` button in the UI kills a task via `POST` to `/v2/tasks/delete`, as opposed to deleting an entire application via `DELETE` to the relevant app endpoint. 

I suspect the difference in signals is down to this.

**Comment by philipnrmn** [comment]
I found some more information in this ticket: #712. It seems that Marathon hands off responsibility for killing tasks to Mesos, which eagerly kills them if they don't terminate within three seconds.

**Comment by jolexa** [comment]
@philipnrmn the DELETE /v2/apps/{appId} endpoint is not the same as /v2/tasks/delete, I'm alittle confused now to be honest based on your comment

What exactly does the UI do when you select **one** app instance checkbox and then hit the kill button?

**Comment by philipnrmn** [comment]
Hi @jolexa, let me try again!

There are two endpoints in question, one for tasks, one for apps. 

`POST /v2/tasks/delete` deletes one or more specified tasks. This is what you see working when you select one app instance checkbox and hit the kill button. 

`DELETE /v2/apps/{appId}` removes an application and also cleans up its tasks. This is what you see working when you hit the red 'Destroy' option on the settings menu. 

The issue is that in the latter case, although the tasks do receive `SIGTERM`, they also receive `SIGKILL` if they do not terminate within three seconds. The ticket I linked to at #712 discusses how we should go about improving this behaviour.

**Comment by jolexa** [comment]
Ah, I see.
```POST /v2/tasks/delete``` vs ```DELETE /v2/apps/{appId}``` vs what I was actually referring to and mistyped initially, ```DELETE /v2/apps/{appId}/tasks/{taskId} ```

It is all really confusing, but I think the moral of the story is that DELETE is destructive (SIGKILL? handed to mesos?) and POST .../delete is not so destructive (SIGTERM? by marathon?)

I'll try to figure out a PR to the docs here, unless someone beats me to it

**Comment by philipnrmn** [comment]
Great thanks, I'll label this ticket `docs` and leave it open then.

**Comment by meichstedt** [comment]
Note: This issue has been migrated to https://jira.mesosphere.com/browse/MARATHON-1950. For more information see https://groups.google.com/forum/#!topic/marathon-framework/khtvf-ifnp8.
